specs/commandline/sign.md

```diff
@@ -29,14 +29,15 @@ Usage:
   notation sign [flags] <reference>
 
 Flags:
-  -e, --expiry duration          optional expiry that provides a "best by use" time for the artifact. The duration is specified in minutes(m) and/or hours(h). For example: 12h, 30m, 3h20m
-  -h, --help                     help for sign
-  -k, --key string               signing key name, for a key previously added to notation's key list.
-  -p, --password string          password for registry operations (default to $NOTATION_PASSWORD if not specified)
-      --plain-http               registry access via plain HTTP
-      --plugin-config strings    {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values
-      --signature-format string  signature envelope format, options: 'jws', 'cose' (default "jws")
-  -u, --username string          username for registry operations (default to $NOTATION_USERNAME if not specified)
+  -e,  --expiry duration          optional expiry that provides a "best by use" time for the artifact. The duration is specified in minutes(m) and/or hours(h). For example: 12h, 30m, 3h20m
+  -h,  --help                     help for sign
+  -k,  --key string               signing key name, for a key previously added to notation's key list.
+  -p,  --password string          password for registry operations (default to $NOTATION_PASSWORD if not specified)
+       --plain-http               registry access via plain HTTP
+       --plugin-config strings    {key}={value} pairs that are passed as it is to a plugin, refer plugin's documentation to set appropriate values
+       --signature-format string  signature envelope format, options: 'jws', 'cose' (default "jws")
+  -u,  --username string          username for registry operations (default to $NOTATION_USERNAME if not specified)
+  -um, --user-metadata strings    {key}={value} pairs that are added to the signature
```

Comment:
Is this metadata added to the annotation attributes of payload https://github.com/notaryproject/notaryproject/blob/main/specs/signature-specification.md#payload?

Reply:
yes, metadata will be added to the annotation attributes of the payload. is this something that needs to be called out in the spec?

Reply:
I suggest adding a reference to the signature spec, so that people can understand these metadata will be added as annotations for the payload as described in the signature spec payload section. You may be aware, there are also annotation attributes for signature manifest. Check this out Check out this https://github.com/notaryproject/notaryproject/blob/main/specs/signature-specification.md#storage. It may cause confusion.

Maybe we can also make the description of this new flag more clear
`{key}={value} pairs that are added to the signature` -> `{key}={value} pairs that are added to the signature payload`

Reply:
I would avoid any references in command help because the url might change but yes we should add some information about how they are being stored in the spec.

Reply:
not adding a reference to the annotation / signature specs then